server/sonar-server/src/main/java/org/sonar/server/computation/scm/Changeset.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.scm;
+
+import java.util.Objects;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public final class Changeset {
+
+  private final String revision;
+  private final long date;
+  @CheckForNull
+  private final String author;
+
+  private Changeset(Builder builder) {
+    this.revision = builder.revision;
+    this.author = builder.author;
+    this.date = builder.date;
+  }
+
+  public static Builder newChangesetBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private String revision;
+    private long date;
```

Comment:
Builder should use a `Long` date field and detect that none has been set rather than using date `0L` silently